lib/condor.py

```diff
@@ -93,44 +94,59 @@ def load_submit_file(filename):
     return htcondor.Submit(res), nqueue
 
 
-def submit(f, vargs, schedd_add, cmd_args=[]):
+def submit(f, vargs, schedd_args, cmd_args=[]):
```

Comment:
So I thought about this further and discussed briefly with Kevin, and I think we're almost there with this.  Admittedly, passing the `schedd_args` and then extracting the name, as I asked you to do previously, is a little inelegant, especially since we have the `schedd_name` in both the `condor_submit` and `jobsub_submit` cases.  It'd be much better to just pass in the `schedd_name` if possible.

Given that, since the only difference between how we use the `schedd_name` comes from either using `--spool` or not in `condor_submit`, would it be possible to do the following:

1)  Change the signature here for `submit` to take `schedd_name` instead of `schedd_args`.
2)  Add to the signature a `spool` boolean argument (and default it to `False`)
3)  Take lines 129-132 from `condor_submit` and move them into this function, so we construct the `schedd_args` here.  Of course, you'd need to run the check on `spool`, not `args.spool` in that case.
4)  Change the calls to `submit` in `condor_submit` and `jobsub_submit` to pass in the `schedd_name`, and in the case of `condor_submit`, `args.spool`.

I think that'll clean this bit of code up, and help us transition to HTCondor Python bindings in the future, since we'll be passing in information, rather than strings of commands.

Please let me know if you have any questions or concerns about this.

Reply:
I made the changes. And I added in the error messages if a user tries to specify the schedd when using condor_submit

Reply:
Looks good - thanks!